api-matlab/wrapper_physiobatch.m

```diff
@@ -0,0 +1,111 @@
+        % respiration = input('Specify file path to the respiration file (in template starting with RespData_)', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {cardiac};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {respiration};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''};
+     elseif strcmp(data_vendor, 'Siemens_HCP')
+        [physiolog, ~] = spm_select(1,'any','Specify physio data file (sth like tfMRI_MOTOR_LR_Physio_log.txt)',{}, default_path);
+        % physiolog = input('Specify file path to the tfMRI_MOTOR_LR_Physio_log.txt file', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {physiolog};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {physiolog};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''};
+     elseif strcmp(data_vendor, 'Siemens')
+        [physiolog, ~] = spm_select(1,'.*.ecg','Specify physio data file (siemens_PAV.ecg)',{}, default_path);
+        % physiolog = input('Specify file path to the siemens_PAV.ecg file', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {physiolog};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {''};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {''}; 
+     elseif strcmp(data_vendor, 'Siemens_Tics')
+        [cardiac, ~] = spm_select(1,'.*.log','Specify cardiac file (..._PULS.log)',{}, default_path);
+        % cardiac = input('Specify file path to the cardiac file (in template ending with _PULS.log)', 's');
+        [respiration, ~] = spm_select(1,'.*.log','Specify respiration file (..._RESP.log)',{}, default_path);
+        % respiration = input('Specify file path to the respiration file (in template ending with _RESP.log)', 's');
+        [timing, ~] = spm_select(1,'.*.log','Specify scan timing file (..._Info.log)',{}, default_path);
+        % timing = input('Specify file path to the scan timing file (in template ending with _Info.log)', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {cardiac};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {respiration};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {timing};
+    else
+        ME = MException('error:wronginput', 'Wrong vendor specified. Please specify one of BIDS, Philips, Biopac_Txt, GE, Siemens, Siemens_HCP, Siemens_Tics');
+        throw(ME);
+    end
+    
+    % read scan parameters
+    input_params = inputdlg({'Number of scans:','Number of slices:','Number of dummies:', 'TR:', 'Onset slice:'}, 'Parameters for physio');
+    
+    % scan parameters
+    matlabbatch{1}.spm.tools.physio.log_files.sampling_interval = [];
+    matlabbatch{1}.spm.tools.physio.log_files.relative_start_acquisition = 0;
+    matlabbatch{1}.spm.tools.physio.log_files.align_scan = 'last'; % sometimes first, sometimes last
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nslices = str2double(input_params{2});
```

Comment:
Does Physio create RETROICOR regressors per slice? Why is the number of slices important? Is slice order and/or interleaved specifications also important?